### PR TITLE
Reduce the complexity of other/scoring_algorithm.py

### PR DESCRIPTION
**Pull request by MaximSmolskiy**
### Describe your change:

Reduce the complexity of other/scoring_algorithm.py from 16 to 10

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by CaedenPH — COMMENTED** [review]
Maybe I am misunderstanding something, but why is the max complexity being increased?

**Comment by MaximSmolskiy** [comment]
> Maybe I am misunderstanding something, but why is the max complexity being increased?

It was decreased in #7954, but now there are algorithms with greater complexity.

**Comment by MaximSmolskiy** [comment]
@cclauss please review

File: other/scoring_algorithm.py [hunk 1]
```diff
@@ -72,14 +72,45 @@ def procentual_proximity(
...
+    >>> procentual_proximity([[20, 60, 2012],[23, 90, 2015],[22, 50, 2011]], [0, 0, 1])
+    [[20, 60, 2012, 2.0], [23, 90, 2015, 1.0], [22, 50, 2011, 1.3333333333333335]]
+    """
+
+    data_lists: list[list[float]] = get_data(source_data)
+
+    score_lists: list[list[float]] = calculate_each_score(data_lists, weights)
+
+    final_scores: list[float] = generate_final_scores(score_lists)
+
```
Comment (dhruvmanila) [inline other/scoring_algorithm.py:113]:
```suggestion
    data_lists = get_data(source_data)
    score_lists = calculate_each_score(data_lists, weights)
    final_scores = generate_final_scores(score_lists)

```
There's no need to specify variable type hint as the functions are already annotated.

**Review by dhruvmanila — CHANGES_REQUESTED** [review]

File: other/scoring_algorithm.py (hunk 1 above)
Comment (MaximSmolskiy) [reply other/scoring_algorithm.py]: Fixed

**Review by dhruvmanila — APPROVED** [review]

**Pull request merged**